### PR TITLE
add more contract api tests

### PR DESCRIPTION
**Pull request by ailisp**
#764

### CODE REVIEW

File: chain-signatures/contract/src/lib.rs [hunk 1]
```diff
@@ -215,10 +215,10 @@ impl VersionedMpcContract {
         0
     }
 
-    /// This experimantal function calculates the fee for a signature request.
+    /// This experimental function calculates the fee for a signature request.
     /// The fee is volatile and depends on the number of pending requests.
     /// If used on a client side, it can give outdate results.
-    pub fn experimantal_signature_deposit(&self) -> u128 {
```
Comment (volovyks) [inline chain-signatures/contract/src/lib.rs:221]: Thank you! I think I've made that mistake in docs as well.

File: chain-signatures/contract/tests/user_views.rs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    let key: String = contract.view("public_key").await.unwrap().json().unwrap();
+    println!("{:?}", key);
+    let pk = PublicKey::from_str(&key)?;
+    assert_eq!(pk.curve_type(), CurveType::SECP256K1);
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_derived_public_key() -> anyhow::Result<()> {
```
Comment (volovyks) [inline chain-signatures/contract/tests/user_views.rs:35]: We will need to execute actual derivation logic in such a test. Not a blocker now.

File: chain-signatures/contract/tests/vote.rs [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_vote_pk() -> anyhow::Result<()> {
+    let (_, contract, accounts, _) = init_env().await;
+
+    let key: String = contract.view("public_key").await.unwrap().json().unwrap();
+
+    let execution = accounts[2]
```
Comment (volovyks) [inline chain-signatures/contract/tests/vote.rs:207]: We should check that random acc can not vote for pk

**Review by volovyks — APPROVED** [review]
Great work!

File: chain-signatures/contract/tests/user_views.rs (hunk 2 above)
Comment (ppca) [reply chain-signatures/contract/tests/user_views.rs:35]: agree. Probably good to be able to fix the pk in init_env() so we can test the value of the derived pk is correct.

**Review by ppca — APPROVED** [review]

**Comment by volovyks** [comment]
@ailisp merging, lets address all the comments if a follow up PR

**Pull request merged**